### PR TITLE
Avoid Symfony 5.1 deprecations

### PR DESCRIPTION
**Pull request by Jean85**

### CODE REVIEW

File: src/Capsule/Collection.php [hunk 1]
```diff
@@ -221,7 +221,7 @@ private function prepareQuery(string $method, $filters = null, $data = null, arr
         );
 
         $event = new QueryEvent($query);
-        if (class_exists(LegacyEventDispatcherProxy::class)) {
+        if (! class_exists(\Symfony\Component\EventDispatcher\Event::class) || class_exists(LegacyEventDispatcherProxy::class)) {
```
[the remark this answers is not in the document]
Comment (ilario-pierbattista) [reply src/Capsule/Collection.php:224]:
```suggestion
        // The Event class has been removed in Symfony 5, so its absence is used as a trigger to stop using the LegacyEventDispatcherProxy.
        if (! class_exists(\Symfony\Component\EventDispatcher\Event::class) || class_exists(LegacyEventDispatcherProxy::class)) {
```

Right, but I think that these considerations should be kept near the code.

File: src/DependencyInjection/MongoDbBundleExtension.php [hunk 2]
```diff
@@ -118,4 +121,17 @@ private function defineDriverOptionsFactory(array $config)
     {
         return isset($config['driverOptions']) ? new Reference($config['driverOptions']) : null;
     }
+
+    /**
+     * This is needed to avoid the EventDispatcher deprecation from 4.3
+     */
+    private function decorateEventDispatcher(): void
+    {
+        if (class_exists(\Symfony\Component\EventDispatcher\Event::class) && class_exists(LegacyEventDispatcherProxy::class)) {
```
[the remark this answers is not in the document]
Comment (ilario-pierbattista) [reply src/DependencyInjection/MongoDbBundleExtension.php:130]:
```suggestion
        // This condition then is triggered only in Symfony 4.3 and 4.4
        if (class_exists(\Symfony\Component\EventDispatcher\Event::class) && class_exists(LegacyEventDispatcherProxy::class)) {
```

File: .travis.yml [hunk 3]
```diff
@@ -68,7 +68,7 @@ jobs:
     - stage: "Analysis"
       name: "phpstan"
       script:
-        - docker-compose run --no-deps --rm php bin/phpstan analyze -l7 src/ tests/
+        - docker-compose run --no-deps --rm php bin/phpstan analyze --memory-limit -l7 src/ tests/
```
Comment (ilario-pierbattista) [inline .travis.yml:71]: `--memory-limit` requires an argument. What is its purpose here?

File: src/Capsule/Collection.php [hunk 4]
```diff
@@ -221,7 +221,8 @@ private function prepareQuery(string $method, $filters = null, $data = null, arr
         );
 
         $event = new QueryEvent($query);
-        if (class_exists(LegacyEventDispatcherProxy::class)) {
+        // The Event class has been removed in Symfony 5, so its absence is used as a trigger to stop using the LegacyEventDispatcherProxy.
+        if (! class_exists(\Symfony\Component\EventDispatcher\Event::class) || class_exists(LegacyEventDispatcherProxy::class)) {
```
Comment (ilario-pierbattista) [inline src/Capsule/Collection.php:225]: This condition is often repeated. Is it possible to encapsulate it in an external static function?

File: .travis.yml (hunk 3 above)
Comment (Jean85) [reply .travis.yml:71]: I didn't know it required it! It probably has a `-1` default, because putting it there fixed the build.

**Comment by Jean85** [comment]
@ilario-pierbattista fixed for good this time!

**Pull request merged**